### DISCUSSION TITLE
What is the fullform of LGTM?

### DISCUSSION THREAD

**Discussion by Frenemee2020** [github/securitylab#166] [General] [answered]
Does LGTM analyze security issues and also supports any DAST tools?

**Comment by anticomputer** [comment, marked as answer]
LGTM.com (Looks Good To Me) is a SAST based code analysis tool. The technology behind it is called [CodeQL](https://securitylab.github.com/tools/codeql) which [creates a database](https://help.semmle.com/codeql/codeql-cli/procedures/create-codeql-database.html) that represents the code you are analyzing. It includes a full representation of the AST as well as a data flow graph and many other enrichments which you can then query against with the CodeQL [language](https://help.semmle.com/QL/learn-ql/introduction-to-ql.html).

You can use these CodeQL queries to describe and narrow in on known vulnerability patterns in a given database or set of databases. CodeQL comes with an extensive (and always growing) collection of such vulnerability queries built in for all the languages it [supports](https://help.semmle.com/QL/ql-support/index.html). 

LGTM.com is essentially an enterprise frontend to the CodeQL technology, but you can also use it locally through the [CodeQL cli](https://help.semmle.com/codeql/codeql-cli.html) or e.g. the [VSCode extension](https://help.semmle.com/codeql/codeql-for-vscode/procedures/setting-up.html).

Hope that helps and if you have any more questions please do let us know!
